### ISSUE TITLE
JUser::save() deletes all custom user fields values.

### ISSUE THREAD

**Issue by Denitz** [joomla/joomla-cms#17801]
### Steps to reproduce the issue
Create custom user fields.
Edit user and fill any data in custom fields.

Execute:
```
$user = JFactory::getUser($USER_ID);
$user->save();
```

### Expected result

User custom fields values stay same.

### Actual result

All user custom fields are empty (all data from `#__fields_values` table is deleted for this user).

### Additional comments

All other API methods using `JUser::save()` are affected, e.g.:

```
JUserHelper::addUserToGroup();
JUserHelper::removeUserFromGroup();
```

PS. It also looks like enormous amount of queries to fields tables are executed if you have lots fields.

**Comment by ghost** [comment]
@laoneo can you please have a Look?<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/17801">issues.joomla.org/tracker/joomla-cms/17801</a>.</sub>

**Comment by ghost** [comment]
closed as having Pull Request #17837<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/17801">issues.joomla.org/tracker/joomla-cms/17801</a>.</sub>
